### PR TITLE
Add configuration for choosing the branches for which the hook commands should be executed

### PR DESCRIPTION
**Pull request by mikailcf**
## Description
Adding a simple configuration field for when you don't want to run the configured hooks for every branch.

There's a check verifying if the hook commands should be executed for the current working branch.

## How to use
Add the key `active-branches` and set an array value with the name of the branches for which you want the hooks executed.

Example:
```
// package.json
{
  "husky": {
    "hooks": {
      "pre-commit": "npm test",
      "pre-push": "npm test",
      "active-branches": ["master", "staging"],
      "...": "..."
    }
  }
}
```

### CODE REVIEW

File: src/runner/index.ts [hunk 1]
```diff
@@ -35,6 +35,17 @@ export default async function run(
   const oldCommand: string | undefined =
     pkg && pkg.scripts && pkg.scripts[hookName.replace('-', '')]
 
+  const activeBranches: string | undefined =
```
Comment (pamepeixinho) [inline src/runner/index.ts]: Maybe it can be an array of strings. What do you think?

Comment (mikailcf) [reply src/runner/index.ts]: It should indeed be an array, that code would result in an error. Thanks for pointing it out!

**Comment by beppe9000** [comment]
Is it possible instead to expand this in order to specify different scripts for different branches ?

e.g.

```
// package.json
{
  "husky": {
    "hooks": {
      "pre-commit": "npm test",
      "pre-push": "npm test",
      "active-branches": ["master", "staging"],
      "branch-scripts": {
          "master": {
                           "pre-commit": "ng lint",
                           "pre-push": "npm test"
                         }
          "staging": {
                           "pre-commit": "ng lint",
                           "pre-push": "npm xyz"
                         }
       }
      "...": "..."
    }
  }
}
```

**Comment by muntasirsyed** [comment]
Love this! I do agree with @beppe9000 that configuration per branch would perhaps be slightly more useful for most use cases.

**Comment by beppe9000** [comment]
is this implemented ?

**Pull request closed without merging**